pkg/daemon/certs/wicd_cert_manager.go

```diff
@@ -0,0 +1,145 @@
+//go:build windows
+
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package certs
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/openshift/windows-machine-config-operator/pkg/rbac"
+	certificatesv1 "k8s.io/api/certificates/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/certificate"
+)
+
+// WICDCertificateManager manages WICD node certificates
+type WICDCertificateManager struct {
+	nodeName                string
+	certDir                 string
+	bootstrapKubeconfigPath string
+	apiServer               string
+	certDuration            time.Duration
+	started                 bool
+
+	// Certificate paths
+	currentCertPath string
+	currentKeyPath  string
+
+	// Built-in certificate manager
+	certManager certificate.Manager
+}
+
+// NewWICDCertificateManager creates a new WICD certificate manager
+func NewWICDCertificateManager(nodeName, certDir, bootstrapKubeconfigPath, apiServer string, certDuration time.Duration) *WICDCertificateManager {
+	return &WICDCertificateManager{
+		nodeName:                nodeName,
+		certDir:                 certDir,
+		bootstrapKubeconfigPath: bootstrapKubeconfigPath,
+		apiServer:               apiServer,
+		certDuration:            certDuration,
+		currentCertPath:         filepath.Join(certDir, rbac.WICDCertNamePrefix+"-current.pem"),
+		currentKeyPath:          filepath.Join(certDir, rbac.WICDCertNamePrefix+"-current.pem"),
+	}
+}
+
+// StartCertificateManagement starts certificate management for WICD.
+// The certificate manager automatically handles renewal at 80% of certificate lifetime using built-in goroutines.
+// NOTE: Kubernetes does not support certificate revocation for CSR-issued certificates.
+// Once issued, certificates remain valid until expiration. For security, certificates should have
+// short lifetimes and be rotated frequently.
+func (wcm *WICDCertificateManager) StartCertificateManagement(ctx context.Context) error {
+	if wcm.started {
+		return nil
+	}
+
+	// Load bootstrap kubeconfig for initial authentication
+	bootstrapConfig, err := clientcmd.BuildConfigFromFlags("", wcm.bootstrapKubeconfigPath)
+	if err != nil {
+		return fmt.Errorf("failed to load bootstrap kubeconfig from %s: %w", wcm.bootstrapKubeconfigPath, err)
+	}
+
+	defaultConfig := &rest.Config{
```

Comment:
Seems like this should be constructed as part of the New() method. Much the values in the wcm struct are used solely to create this config. Moving this to New() would simplify the struct.